### PR TITLE
feat(surveys): `posthog-js` changes to support valid HTML question rendering

### PR DESCRIPTION
**Pull request by dmarticus**
## Changes

Modified the survey renderer to differentiate between `Text` and `HTML` references when writing question descriptions.  Before this change, we had a bug where users could bypass the HTML question description markup paywall by writing raw HTML into their Text description (and it would still be interpolated as HTML if the tags were valid), so this change makes it so that we only interpolate HTML in the popup survey if the HTML feature is (a) paid for and (b) enabled.

Needs to ship with the associated `posthog` changes here: https://github.com/PostHog/posthog/pull/22533

Closes: https://github.com/PostHog/posthog/issues/22509

...

## Checklist
- [x] Tests for new code (see [advice on the tests we use](https://github.com/PostHog/posthog-js#tiers-of-testing))
    - Added some tests for this new HTML rendering behavior (the tests check if HTML is rendered if the appropriate content types are set, and not rendered if not)
- [x] Accounted for the impact of any changes across different browsers
- [x] Accounted for backwards compatibility of any changes (no breaking changes in posthog-js!)
    - For any surveys missing this contentType field, I'm doing the legacy behavior and treating them like HTML

### CODE REVIEW

File: src/extensions/surveys/components/ConfirmationMessage.tsx [hunk 1]
```diff
@@ -26,13 +28,22 @@ export function ConfirmationMessage({
...
-                    {confirmationDescription && (
-                        <div
-                            style={{ color: textColor }}
-                            className="thank-you-message-body"
-                            dangerouslySetInnerHTML={{ __html: confirmationDescription }}
-                        />
-                    )}
+                    {confirmationDescription ? (
+                        confirmationDescriptionContentType === 'text' ? (
+                            <div style={{ color: textColor }} className="thank-you-message-body">
```
Comment (neilkakkar) [inline src/extensions/surveys/components/ConfirmationMessage.tsx:33]: I'd love this to be a helper so we don't have so much duplication here, this is very prone to errors. Something like:

`renderChildrenAsTextOrHtml({ component, children, renderAsHtml }) => // conditionally either add `dangerouslySetInnerHTML` on the component, or just add children as a child node`

Think you can use cloneelement for this https://preactjs.com/guide/v10/api-reference/#cloneelement - unless you find a better way ;)

File: src/posthog-surveys-types.ts [hunk 2]
```diff
@@ -43,9 +44,12 @@ export enum SurveyType {
 
 export type SurveyQuestion = BasicSurveyQuestion | LinkSurveyQuestion | RatingSurveyQuestion | MultipleSurveyQuestion
 
+export type SurveyQuestionDescriptionContentType = 'html' | 'text'
+
 interface SurveyQuestionBase {
     question: string
     description?: string | null
+    descriptionContentType?: 'html' | 'text'
```
Comment (neilkakkar) [inline src/posthog-surveys-types.ts:52]:
```suggestion
    descriptionContentType?: SurveyQuestionDescriptionContentType
```

**Review by neilkakkar — APPROVED** [review]
I'm a bit bummed this doesn't solve it for other fields 😅, so we should eventually revisit this, but also not the highest priority so agree with your balancing here.

 This looks good to me - just 2 comments :)

File: src/extensions/surveys/components/ConfirmationMessage.tsx (hunk 1 above)
Comment (dmarticus) [reply src/extensions/surveys/components/ConfirmationMessage.tsx]: wrote a helper method for this that I used in both instances, seemed to work great!

Comment (neilkakkar) [reply src/extensions/surveys/components/ConfirmationMessage.tsx]: Amazing, we can reuse this now for when we extend this to other fields 🚀

**Comment by neilkakkar** [comment]
annoying how comments work on commits -> https://github.com/PostHog/posthog-js/commit/89e6d4767fb6272d00447453a13f9acac95ea746#r142799078

**Pull request merged**